From 51a7fea9b737ec81ed2d41c4847976e3dfa9bff4 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Thu, 30 Jan 2025 15:46:11 +0100
Subject: [PATCH 1483/1695] core: psa-adac: fix PSA_ADAC_LOG_(T)DUMP with
 snprintf instead of sprintf

sprintf is deprecated. Use snprintf instead. The maximum bytes to be
written in _tmp buffer is 3, the two characters for hexadecimal value %02x
plus the null character.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 03edc3fd40074fe8d300bfab92157c5773f04e79 core: psa-adac: fix PSA_ADAC_LOG_(T)DUMP with snprintf instead of sprintf]
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Change-Id: I0182cf9f3374f9b1f8408f8a64f6d66a13464c93
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/433583
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/lib/psa-adac/psa-adac/core/include/psa_adac_debug.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/lib/psa-adac/psa-adac/core/include/psa_adac_debug.h b/core/lib/psa-adac/psa-adac/core/include/psa_adac_debug.h
index 5f8e32f17..c5281d2ca 100644
--- a/core/lib/psa-adac/psa-adac/core/include/psa_adac_debug.h
+++ b/core/lib/psa-adac/psa-adac/core/include/psa_adac_debug.h
@@ -70,7 +70,7 @@
         for (_i = 0; _i * 16 + _j < _size; _i++, _j = 0) { \
             char _tmp[256] = {0}; \
             for (_j = 0, _k = 0; _i * 16 + _j < _size && _j < 16; _j++) { \
-                _k += sprintf(_tmp + _k, "%02x", ((uint8_t *)_buff)[_i * 16 + _j]); \
+                _k += snprintf(_tmp + _k, 3, "%02x", ((uint8_t *)_buff)[_i * 16 + _j]); \
             } \
             PSA_ADAC_LOG_FUNC_AND_LEVEL("debug", _who); \
             PSA_ADAC_LOG_PRINT("%-10.10s %04x: %s\n", _label, _i * 16, _tmp); \
@@ -89,7 +89,7 @@
         for (_i = 0; _i * 16 + _j < _size; _i++, _j = 0) { \
             char _tmp[256] = {0}; \
             for (_j = 0, _k = 0; _i * 16 + _j < _size && _j < 16; _j++) { \
-                _k += sprintf(_tmp + _k, "%02x", ((uint8_t *)_buff)[_i * 16 + _j]); \
+                _k += snprintf(_tmp + _k, 3, "%02x", ((uint8_t *)_buff)[_i * 16 + _j]); \
             } \
             PSA_ADAC_LOG_FUNC_AND_LEVEL("trace", _who); \
             PSA_ADAC_LOG_PRINT("%-10.10s %04x: %s\n", _label, _i * 16, _tmp); \
-- 
2.47.2

